File: app/src/main/java/com/maherhanna/cheeta/MainActivity.java
package com.maherhanna.cheeta;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.RadioButton;
import android.widget.Spinner;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    Button btn_start;
    RadioButton rb_white;
    RadioButton rb_black;


    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        btn_start = findViewById(R.id.btn_start);
        rb_white = findViewById(R.id.rb_white);
        rb_black = findViewById(R.id.rb_black);

        btn_start.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent gameIntent = new Intent(MainActivity.this,GameActivity.class);
                int humanPiecesColor = Piece.WHITE;
                if(rb_black.isSelected())humanPiecesColor = Piece.BLACK;
                gameIntent.putExtra("humanPiecesColor",humanPiecesColor);
                startActivity(gameIntent);
            }
        });



    }


}

File: app/src/main/java/com/maherhanna/cheeta/MainActivity.kt
package com.maherhanna.cheeta

import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.Button
import android.widget.RadioButton
import androidx.appcompat.app.AppCompatActivity
import com.google.android.gms.ads.AdError
import com.google.android.gms.ads.AdRequest
import com.google.android.gms.ads.FullScreenContentCallback
import com.google.android.gms.ads.LoadAdError
import com.google.android.gms.ads.MobileAds
import com.google.android.gms.ads.interstitial.InterstitialAd
import com.google.android.gms.ads.interstitial.InterstitialAdLoadCallback
import com.maherhanna.cheeta.core.Piece


class MainActivity : AppCompatActivity() {
    lateinit var btn_start: Button
    var rb_white: RadioButton? = null
    lateinit var rb_black: RadioButton
    private var mInterstitialAd: InterstitialAd? = null
    private final var TAG = "MainActivity"
    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        MobileAds.initialize(this) { }

        var adRequest = AdRequest.Builder().build()

        InterstitialAd.load(
            this,
            "ca-app-pub-3940256099942544/1033173712",
            adRequest,
            object : InterstitialAdLoadCallback() {
                override fun onAdFailedToLoad(adError: LoadAdError) {
                    Log.d(TAG, adError.toString())
                    mInterstitialAd = null
                }

                override fun onAdLoaded(interstitialAd: InterstitialAd) {
                    Log.d(TAG, "Ad was loaded.")
                    mInterstitialAd = interstitialAd
                }
            })

        mInterstitialAd?.fullScreenContentCallback = object : FullScreenContentCallback() {
            override fun onAdClicked() {
                // Called when a click is recorded for an ad.
                Log.d(TAG, "Ad was clicked.")
            }

            override fun onAdDismissedFullScreenContent() {
                // Called when ad is dismissed.
                Log.d(TAG, "Ad dismissed fullscreen content.")
                mInterstitialAd = null
            }

            override fun onAdFailedToShowFullScreenContent(adError: AdError) {
                // Called when ad fails to show.
                Log.e(TAG, "Ad failed to show fullscreen content.")
                mInterstitialAd = null
            }

            override fun onAdImpression() {
                // Called when an impression is recorded for an ad.
                Log.d(TAG, "Ad recorded an impression.")
            }

            override fun onAdShowedFullScreenContent() {
                // Called when ad is shown.
                Log.d(TAG, "Ad showed fullscreen content.")
            }
        }
        supportActionBar!!.hide()
        setContentView(R.layout.activity_main)
        btn_start = findViewById<Button>(R.id.btn_start)
        rb_white = findViewById(R.id.rb_white)
        rb_black = findViewById(R.id.rb_black)
        btn_start.setOnClickListener(View.OnClickListener {
            val gameIntent = Intent(this@MainActivity, GameActivityNewUi::class.java)
            var humanPiecesColor = Piece.WHITE
            if (rb_black.isChecked()) humanPiecesColor = Piece.BLACK
            gameIntent.putExtra("humanPiecesColor", humanPiecesColor)
            if (mInterstitialAd != null) {
                mInterstitialAd?.show(this)
            } else {
                Log.d("TAG", "The interstitial ad wasn't ready yet.")
            }
            startActivity(gameIntent)
        })
    }
}